src/sync/sync_session.cpp

```diff
@@ -636,8 +636,15 @@ void SyncSession::handle_error(SyncError error)
             }
             break;
         case NextStateAfterError::inactive: {
-            std::unique_lock<std::mutex> lock(m_state_mutex);
-            advance_state(lock, State::inactive);
+            if (error.is_client_reset_requested()) {
+                std::unique_lock<std::mutex> lock(m_state_mutex);
+                cancel_pending_waits(lock, error.error_code);
```

Comment:
is this the correct error_code that you need at the js level?

Reply:
Yeah, I see your point. This could produce a range of different errors, and narrow it down to just one might be easier for users to react to.